File: app/src/main/java/ml/dvnlabs/animize/fragment/library_n/star.java
package ml.dvnlabs.animize.fragment.library_n;


import android.os.AsyncTask;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.os.Handler;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;


import java.util.ArrayList;


import ml.dvnlabs.animize.R;

import ml.dvnlabs.animize.database.PackageStarDBHelper;
import ml.dvnlabs.animize.database.model.starland;

import ml.dvnlabs.animize.recyclerview.packagelist.starlist_adapter;
import ml.dvnlabs.animize.view.AutoGridLayoutManager;

/**
 * A simple {@link Fragment} subclass.
 */
public class star extends Fragment {

    private PackageStarDBHelper packageStarDBHelper;
    private int counter =0;
    private RecyclerView rv_starred;
    private starlist_adapter adapter;
    private AutoGridLayoutManager LayoutManager;
    private RelativeLayout loading,voided;
    private SwipeRefreshLayout refreshLayout;
    private String tempid;




    public star() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        final View view =inflater.inflate(R.layout.fragment_star, container, false);
        rv_starred = view.findViewById(R.id.star_list);
        voided = view.findViewById(R.id.starred_void);
        /*loading = view.findViewById(R.id.starred_loading);*/
        refreshLayout = view.findViewById(R.id.starred_refresh);
        packageStarDBHelper = new PackageStarDBHelper(getActivity());
        refresh_list();
        swipe_refresh();

        return view;
    }
    private void refresh_list(){
        if (packageStarDBHelper.isAvail()){
            voided.setVisibility(View.GONE);
            getStarredPackage getStarredPackage = new getStarredPackage();
            getStarredPackage.execute();
        }else {
            /*loading.setVisibility(View.GONE);*/
            voided.setVisibility(View.VISIBLE);
        }
    }





    @Override
    public void onResume() {
        super.onResume();
    }

    private void swipe_refresh(){
        refreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                refresh_list();
                // To keep animateView for 4 seconds
                new Handler().postDelayed(new Runnable() {
                    @Override public void run() {
                        // Stop animateView (This will be after 3 seconds)
                        refreshLayout.setRefreshing(false);
                    }
                }, 2000); // Delay in millis
            }
        });

        refreshLayout.setColorSchemeColors(
                getResources().getColor(android.R.color.holo_blue_bright),
                getResources().getColor(android.R.color.holo_green_light),
                getResources().getColor(android.R.color.holo_orange_light),
                getResources().getColor(android.R.color.holo_red_light)
        );
    }
    private class getStarredPackage extends AsyncTask<String,Void, ArrayList<starland>>{
        @Override
        protected void onPreExecute(){

        }
        @Override
        protected ArrayList<starland> doInBackground(String... params){

            return packageStarDBHelper.getStarredList();

        }

        @Override
        protected void onPostExecute(ArrayList<starland> pa){
            counter = pa.size();
            System.out.println("COUNTER: "+counter);
            if (!pa.isEmpty()){
                LayoutManager = new AutoGridLayoutManager(getContext(),500);
                adapter = new starlist_adapter(pa,getActivity(),R.layout.rv_starredpackage);
                rv_starred.setLayoutManager(LayoutManager);
                rv_starred.setHasFixedSize(true);
                rv_starred.setAdapter(adapter);
            }else{
                voided.setVisibility(View.VISIBLE);
            }
        }

    }
}

File: app/src/main/java/ml/dvnlabs/animize/ui/fragment/library/Star.kt
/*
 * Copyright (c) 2020.
 * Animize Devs
 * Copyright 2019 - 2020
 * Davin Alfarizky Putra Basudewa <dbasudewa@gmail.com>
 * This program used for watching anime without ads.
 *
 */
package ml.dvnlabs.animize.ui.fragment.library

import android.os.Bundle
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import ml.dvnlabs.animize.R
import ml.dvnlabs.animize.database.AnimizeDatabase
import ml.dvnlabs.animize.ui.recyclerview.packagelist.StarListAdapter
import ml.dvnlabs.animize.ui.viewmodel.CommonViewModel
import ml.dvnlabs.animize.view.AutoGridLayoutManager
import org.koin.androidx.viewmodel.ext.android.sharedViewModel
import org.koin.core.KoinComponent
import org.koin.core.inject
import org.koin.core.parameter.parametersOf

class Star : Fragment(), KoinComponent {
    private val animizeDB: AnimizeDatabase by inject { parametersOf(context) }
    private var counter = 0
    private var rvStarred: RecyclerView? = null
    private var adapter: StarListAdapter? = null
    private var autoGridLayoutManager: AutoGridLayoutManager? = null
    private var voided: RelativeLayout? = null
    private var refreshLayout: SwipeRefreshLayout? = null
    private val commonViewModel: CommonViewModel by sharedViewModel()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_star, container, false)
        rvStarred = view.findViewById(R.id.star_list)
        voided = view.findViewById(R.id.starred_void)
        refreshLayout = view.findViewById(R.id.starred_refresh)
        refreshList()
        swipeRefresh()

        commonViewModel.libraryScrolledToTop.observe(viewLifecycleOwner, Observer {
            if (it) {
                scrollToTop()
            }
        })
        return view
    }

    private fun refreshList() = CoroutineScope(Dispatchers.Main).launch {
        if (animizeDB.animeDAO().getAllStarredAnime().isNotEmpty()) {
            voided!!.visibility = View.GONE
            if (adapter != null) {
                adapter!!.notifyDataSetChanged()
            }
            getStarredPackage()
        } else {
            voided!!.visibility = View.VISIBLE
        }
    }


    private fun swipeRefresh() {
        refreshLayout!!.setOnRefreshListener {
            refreshList()
            // To keep animateView for 4 seconds
            Handler().postDelayed({ // Stop animateView (This will be after 3 seconds)
                refreshLayout!!.isRefreshing = false
            }, 2000) // Delay in millis
        }
        refreshLayout!!.setColorSchemeColors(
                ContextCompat.getColor(requireContext(), android.R.color.holo_blue_bright),
                ContextCompat.getColor(requireContext(), android.R.color.holo_green_light),
                ContextCompat.getColor(requireContext(), android.R.color.holo_orange_light),
                ContextCompat.getColor(requireContext(), android.R.color.holo_red_light)
        )
    }

    private suspend fun getStarredPackage() {
        val pa = animizeDB.animeDAO().getAllStarredAnime()
        autoGridLayoutManager = AutoGridLayoutManager(requireContext(), 500)
        adapter = StarListAdapter(pa, requireContext(), R.layout.rv_starredpackage)
        rvStarred!!.adapter = adapter
        rvStarred!!.layoutManager = autoGridLayoutManager
    }

    private fun scrollToTop() {
        rvStarred!!.smoothScrollToPosition(0)
        commonViewModel.changeLibraryScrolledToTop()
    }
}